app/models/user.rb

```diff
@@ -80,6 +81,10 @@ def self.from_merged_attrs(existing_user, new_user)
     User.new(attrs)
   end
 
+  def self.from_saml(saml_response)
+    User.new(SAML::DeSerializer.new(saml_response).user_attributes)
```

Comment:
Can we call this 

SAML::UserAttributes.new(saml_response)

and have the initializer invoke user_attributes, so you don't have to call it?

Otherwise this plays more like a Decorator than a Deserializer.